### DISCUSSION TITLE
Octave 6.3.0 on macOS 11.5.1 `mkoctfile` contains stale link old gcc 11.1.0 version

### DISCUSSION THREAD

**Discussion by siko1056** [Homebrew/discussions#1986] [Everyday usage]
On my macOS system the `mkoctfile` command from a freshly installed Octave 6.3.0 on macOS 11.5.1 contains a stale link to old gcc 11.1.0 version (current is 11.2.0).  Most likely fix is to rebuild the bottle?  How can I trigger such a rebuild?

Thanks for you help :slightly_smiling_face: 

Some system information:
```
% pwd
/usr/local/Cellar/octave

% grep -r "11\.1\.0" .
Binary file ./6.3.0/bin/mkoctfile matches
Binary file ./6.3.0/bin/mkoctfile-6.3.0 matches
./6.3.0/lib/pkgconfig/octave.pc:Libs.private: -lcurl -lcholmod -lumfpack -lamd -lcamd -lcolamd -lccolamd -lcxsparse -lsuitesparseconfig -larpack -lqrupdate -lfftw3_threads -lfftw3 -lfftw3f_threads -lfftw3f  -L/usr/local/opt/openblas/lib -lopenblas -lreadline -lncurses  -lpcre -ldl   -L/usr/local/Cellar/gcc/11.1.0_1/lib/gcc/11/gcc/x86_64-apple-darwin20/11.1.0 -L/usr/local/Cellar/gcc/11.1.0_1/lib/gcc/11/gcc/x86_64-apple-darwin20/11.1.0/../../.. -lgfortran -lquadmath -lm  -lm           -liconv   -Wl,-framework -Wl,CoreFoundation
Binary file ./6.3.0/lib/octave/6.3.0/liboctinterp.dylib matches
Binary file ./6.3.0/lib/octave/6.3.0/liboctinterp.9.dylib matches
./6.3.0/share/octave/6.3.0/m/time/datevec.m:%!           "EST+5EDT,M3.2.0/2,M11.1.0/2"          ... America/New_York

% ls /usr/local/Cellar/gcc/                                             
11.2.0

% brew info gcc                
gcc: stable 11.2.0 (bottled), HEAD
GNU compiler collection
https://gcc.gnu.org/
/usr/local/Cellar/gcc/11.2.0 (2,163 files, 459.7MB) *
  Poured from bottle on 2021-08-02 at 15:15:28
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gcc.rb
License: GPL-3.0-or-later with GCC-exception-3.1
==> Dependencies
Required: gmp ✔, isl ✔, libmpc ✔, mpfr ✔, zstd ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 107,166 (30 days), 326,615 (90 days), 1,307,159 (365 days)
install-on-request: 47,508 (30 days), 144,248 (90 days), 605,640 (365 days)
build-error: 0 (30 days)
```

**Comment by siko1056** [comment]
Perhaps my question is asked at the wrong location.  Shall I open an issue instead?  I am happy to answer any questions, if my request was confusing.

However, if someone stumbles over the same problem.  A dirty fix is to run the following two lines from the Terminal:
```bash
cd /usr/local/Cellar/gcc/11.2.0/lib/gcc/11/gcc/x86_64-apple-darwin20/
ln -s 11.2.0 11.1.0
```

And a thread with further Octave related discussion: https://octave.discourse.group/t/problems-with-the-installation-of-control-toolbox-of-octave-in-macos-big-sur/1487/20

**Reply by SMillerDev** [reply]
A pull request with a test that fails in this case would be better, to make sure we catch this on the next gcc update.

**Reply by siko1056** [reply]
Thank you for the feedback.  A year ago, there was a field "revision" in the formula to do enforce a rebuild

https://github.com/Homebrew/homebrew-core/blob/master/Formula/octave.rb

This field is gone.  How does a PR look these days just for the sake of rebuilding the formula?

**Reply by SMillerDev** [reply]
The field still exist, it just isn't used if the revision is `0`.

**Reply by siko1056** [reply]
Thanks for the info, I am preparing a PR now :+1:

**Reply by siko1056** [reply]
Done: https://github.com/Homebrew/homebrew-core/pull/84245
